engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/id/execution/IdExecutionInstrument.java

```diff
@@ -66,6 +68,7 @@ private static class IdEventNodeFactory implements ExecutionEventNodeFactory {
     private final CallTarget entryCallTarget;
     private final Callbacks callbacks;
     private final Timer timer;
+    private final Map<Node, ExecutionEnvironment> currentNodeEnvironments;
```

Comment:
- I believe this is a wrong place for the `Map`
- Moreover the key in the map is really weird
- there already is `IdExecutionService.Callbacks.getExecutionEnvironment`
   - that's a "bridge" from instrumentation to the IDE
- the map should be in the `IdExecutionService.Callbacks` implementation
   - the map should be from UUID, not a node

Reply:
Ah, now I remember why I didn't use UUID - sometimes nested nodes inherit ID from their parent which breaks this logic completely.

Reply:
The latter also explains why the _existing_ logic is more complicated than at first sight it needs to be with additional condition checks for `originalExecutionEnvironment`.

Reply:
- yes, the logic of inheriting UUID was there since I joined
- what's wrong with it?
- at most multiple nodes adhere to the same UUID...
- all of them would change the state/environment